### ISSUE TITLE
kodi wayland fullscreen mode

### ISSUE THREAD

**Issue by mihailescu2m** [MaxKellermann/xbmc#5]
I think kodi on wayland does not get properly into fullscreen mode.

On Gnome3, the gnome top panel is still visible, and the bottom part of kodi interface is hidden, since everything is shifted down with the height of the panel.

**Comment by MaxKellermann** [comment]
Kodi calls `wl_shell_surface_set_fullscreen()` which asks the server to make Kodi full-screen, and do whatever it takes. This is what everybody else does as well. If this doesn't hide the Gnome panel, I don't know what else to do.
Can you name another Wayland application which hides the panel when switching to full-screen?

**Comment by mihailescu2m** [comment]
Yes, I've looked for solutions as well, but didn't find any. I tried `WL_SHELL_SURFACE_FULLSCREEN_METHOD_DEFAULT` but of course no change. Maybe I'll try calling `wl_display_sync` but I don't think that will do anything.
Apps that hide the gnome panel that I know of are GTK apps e.g. totem, so it might be a GTK thing that does it but we can't use that.

For now I installed a gnome extension that has "intellihide" - when an app comes over the panel, the panel will automatically hide. Hope this is useful for anyone else looking for a solution.

I don't know what "fullscreen" means for kodi, but in wayland (and in Hybris windowing that i am trying in another build, which runs directly over the android hwcomposer), kodi shows "windowed, 1920x1280 resolution" in the settings. Any idea how why that happens?

**Comment by MaxKellermann** [comment]
In our Wayland code, there is only full-screen, and windowed mode simply is not implemented. I guess the indicator is wrong, because (maybe) our Wayland code doesn't report the current state properly.
